lib/Driver/Compilation.cpp

```diff
@@ -631,8 +631,15 @@ int Compilation::performSingleCommand(const Job *Cmd) {
   const char *ExecPath = Cmd->getExecutable();
   const char **argv = Argv.data();
 
-  for (auto &envPair : Cmd->getExtraEnvironment())
+  for (auto &envPair : Cmd->getExtraEnvironment()) {
+#if defined(_MSC_VER)
+    std::string envStr = envPair.first;
```

Comment:
Nitpick: please use an llvm::SmallString instead of std::string, so that smaller settings don't require an extra heap allocation.